backend/sftp/file.go

```diff
@@ -379,12 +379,41 @@ func (f *File) String() string {
 // openFile wrapper allows us to inject a file opener (for mocking) vs the defaultOpenFile.
 func (f *File) openFile(flags int) (ReadWriteSeekCloser, error) {
 	if f.sftpfile != nil {
-		FlagsUsedContainsReadOnly := f.flagsUsed&(os.O_WRONLY|os.O_RDWR) == 0
-		flagsContainsReadWrite := flags&os.O_RDWR != 0
+
+		// this case shouldn't normally exist except when we've set our own ReadWriteSeekCloser in tests
```

Comment:
This was only an issue because the read-only flag `os.O_RDONLY` is equal to the zero-value of the struct member `flagsUsed`.  It is only relevant in tests where we've set our own ReadWriteSeekCloser but have never called a read action.